### ISSUE TITLE
Gdrive - download file docs differ from code example here and additional question

### ISSUE THREAD

**Issue by Mk-Etlinger** [google/google-api-nodejs-client#1233]
Hey all, this library is super awesome and has helped me out so much! Thanks for that!

Here's the snippet on google docs(I know you can't do much about this):

https://developers.google.com/drive/api/v3/manage-downloads
```
var fileId = '1ZdR3L3qP4Bkq8noWLJHSr_iBau0DNT4Kli4SxNc2YEo';
var dest = fs.createWriteStream('/tmp/resume.pdf');
drive.files.export({
  fileId: fileId,
  mimeType: 'application/pdf'
})
    .on('end', function () {
      console.log('Done');
    })
    .on('error', function (err) {
      console.log('Error during download', err);
    })
    .pipe(dest);
```

In this example: https://github.com/google/google-api-nodejs-client/blob/master/samples/drive/export.js

I notice that you're using a promise, but also you're `await`ing the initial response.

So two questions:

1. Does using `await response = drive.files.export()` store the binary data in memory?(so if 1,000 users go to download a 1GB file, will it crash the server?)

2. Why do you wrap it in a Promise? Is it for error handling and if so, why not use a Try...Catch?

Thanks for any insight into this, much appreciated.

**Comment by JustinBeckwith** [comment]
Greetings!  The sample at https://developers.google.com/drive/api/v3/manage-downloads is wrong 🤦‍♂️ .  This library used to directly return a stream in some cases, or a callback with data.  We now always invoke the callback (or promise) when returning data.  So to answer your questions:

> Does using await response = drive.files.export() store the binary data in memory?(so if 1,000 users go to download a 1GB file, will it crash the server?)

Nope!  Even after the `await`, you're still just getting a reference to the stream.  It hasn't completed streaming anything yet (as you can see in the sample);

> Why do you wrap it in a Promise? Is it for error handling and if so, why not use a Try...Catch?

We transitioned from `request` over to `axios` for handling our underlying http requests.  `axios` uses a promise style API, similar to what you'd find on the client side with fetch. While we could return a stream synchronously, it's a lot more consistent to always return data (whether in stream or JSON form) after the initial promise resolves. 

TL:DR: the samples in this repo are always going to be more up to date. 

Hope this helps!

**Comment by Mk-Etlinger** [comment]
@JustinBeckwith Thank you so much! I see what you mean about returning the ref to the stream, it makes a lot more sense that way. I was able to figure out some of the underlying mechanisms of the Node stream while I was researching but I really appreciate your time and response.

Our team determined that this was actually not the ideal way to force a download to the client(we were trying to avoid the dreaded gdrive 403 error for people logged into multiple accounts) and so we ended up with a simple fix that involved updating the webContentLink(adding `a/appDomain.com`). Even though this wouldn't directly crash the server via memory, it would increase the bandwidth and decrease download speeds to people abroad. In case you were curious :)

Have a great day sir!
